libs/components/editor-core/src/editor/editor.ts

```diff
@@ -333,6 +333,12 @@ export class Editor implements Virgo {
         return await this.getBlock({ workspace: this.workspace, id: blockId });
     }
 
+    async getBlockByIds(ids: string[]): Promise<Awaited<AsyncBlock | null>[]> {
+        return await Promise.all(
+            ids.map(id => this.getBlock({ workspace: this.workspace, id }))
+        );
+    }
```

Comment:
Add filter to simplify return types

```suggestion
    async getBlockByIds(ids: string[]): Promise<Awaited<AsyncBlock>[]> {
        return await Promise.all(
            ids.map(id => this.getBlock({ workspace: this.workspace, id }))
        ).filter(Boolean);
    }
```

Reply:
getBlockByIds maybe use in multiple places, uncoupled view logic in it